### PR TITLE
Add tombstone field to actor_definitions table

### PR DESCRIPTION
**Pull request by pmossman**
## What
https://github.com/airbytehq/airbyte/issues/9912 was opened because connector definition deletions via API were not persisting. I added the ability to delete connector definitions while the database denormalization project was ongoing, and the new 'tombstone' field did not make it into the actor_definitions table.

This PR adds a migration to add the tombstone column, and updates configPersistence to write handle the new column.

## How
- Migration adds 'tombstone' boolean column to actor_definition table
- Modified YamlSeedConfigPersistence to set the tombstone column when loading
  - Without this, ArchiveHandler tests fail because yaml persistence and db persistence exports are directly compared
- Added a bunch of .withTombstone(false) to tests in order to get them passing

## 🚨 User Impact 🚨
This should fix connector definition deletions

### CODE REVIEW

File: airbyte-db/lib/src/test/java/io/airbyte/db/instance/configs/migrations/V0_35_14_001__AddTombstoneToActorDefinitionTest.java
Comment (subodh1810) [inline airbyte-db/lib/src/test/java/io/airbyte/db/instance/configs/migrations/V0_35_14_001__AddTombstoneToActorDefinitionTest.java:19]: Can we populate the table before hand with a random record and then run the migration and then assert the value of the new column to make sure the not null constraint is being respected.

**Review by subodh1810 — COMMENTED** [review]
Can you update the schema of the table in the file 
`airbyte-db/lib/src/main/resources/configs_database/normalized_tables_schema.txt` 

Also once you merge this, when you make a release, you will have to upgrade the `CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION` parameter in all the places to the version introduced.  

When this would be released to cloud, again `CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION` parameter would need to be updated before deploying as part of the upgrade OSS version PR. 

This is manual right now

File: airbyte-db/lib/src/test/java/io/airbyte/db/instance/configs/migrations/V0_35_14_001__AddTombstoneToActorDefinitionTest.java
Comment (pmossman) [reply airbyte-db/lib/src/test/java/io/airbyte/db/instance/configs/migrations/V0_35_14_001__AddTombstoneToActorDefinitionTest.java:24]: Good idea, added this!

File: airbyte-db/lib/src/main/resources/configs_database/normalized_tables_schema.txt
Comment (pmossman) [inline airbyte-db/lib/src/main/resources/configs_database/normalized_tables_schema.txt:80]: @subodh1810 added

**Review by subodh1810 — APPROVED** [review]
Nice! Remember to update the `CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION` at all the places and when you deploy this to cloud, there as well

**Review by cgardens — APPROVED** [review]

**Pull request merged**
